### ISSUE TITLE
Support for worktrees should NOT be identical to the support for submodules

### ISSUE THREAD

**Issue by PawelLipski** [PawelLipski/git-machete#360]
As noted by @kgadek:

```
❯ git worktree add ../foobar some-branch
Preparing worktree (checking out 'some-branch')
HEAD is now at 93fa883a .....

❯ cd ../foobar

❯ git machete status --list-commits-with-hashes

No branches listed in /Users/kgadek/src/some-project/.git/worktrees/foobar/machete; use git machete discover or git machete edit, or edit /Users/kgadek/src/some-project/.git/worktrees/foobar/machete manually.

❯ file .git
.git: ASCII text

❯ cat .git
gitdir: /Users/kgadek/src/some-project/.git/worktrees/foobar
```

This is exactly what happens for submodules (`machete` file is sought for in per-submodule `.git/` folder, and NOT in the top-project `.git/` folder)... but given the different nature of worktrees, we should rather look for `machete` file in top-project `.git/` folder (so `/Users/kgadek/src/some-project/.git/machete` and not `/Users/kgadek/src/some-project/.git/worktrees/foobar/machete`).

TBD how to elegantly distinguish between the submodule and worktree case (probably just whether the git path ends with `/.git/worktrees/<folder>`?)

Also, important to make sure that git compatibility is preserved (shouldn't be very difficult tho).

**Comment by PawelLipski** [comment]
See the very first issue in this repo for handling of submodules: https://github.com/VirtusLab/git-machete/issues/1

**Comment by kgadek** [comment]
Looks fixed in 3.6.1 🎉

**Comment by PawelLipski** [comment]
Right! I forgot to close this issue
